cmd/cstor-pool-mgmt/volumereplica/volumereplica.go

```diff
@@ -669,3 +671,172 @@ func GetAndUpdateReplicaID(cvr *apis.CStorVolumeReplica) error {
 	}
 	return nil
 }
+
+// GetAndUpdateSnapshotInfo get the snapshot list from ZFS and updates in CVR status.
+// Execution happens in following steps:
+// 1. Get snapshot list from ZFS
+// 2. Checks whether above snapshots exist on CVR under Status.Snapshots:
+//    2.1 If snapshot doesn't exist then get the info of snapshot from ZFS and update
+//        the details in CVR.Status.Snapshots
+// 3. Verify and delete the snapshot details on CVR if it is deleted from ZFS
+// 4. Update the pending list of snapshots by verifying with snapshot list obtained from step1
+// 5. If replica is under rebuilding get the snapshot list from peer CVR and update them
+//    under pending snapshot list
+func GetAndUpdateSnapshotInfo(
+	clientset clientset.Interface, cvr *apis.CStorVolumeReplica) error {
+	volName := cvr.GetLabels()[string(apis.PersistentVolumeCPK)]
+	dsName := PoolNameFromCVR(cvr) + "/" + volName
+
+	snapList, err := GetSnapshotList(dsName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get the list of snapshots")
+	}
+
+	// Add/Delete a snapshot if it doesn't exist in CVR but exists in ZFS
+	err = addOrDeleteSnapshotListInfo(cvr, snapList)
```

Comment:
you can call it as updateSnapshotListInfo

Reply:
I am bit worried to use term `update` which means people can understand from function name that we might have doing etcd update.